src/routes/projects.rs

```diff
@@ -48,34 +46,11 @@ pub async fn projects_get(
 
     let user_option = get_user_from_headers(req.headers(), &**pool).await.ok();
 
-    let mut projects = Vec::new();
-
-    for project_data in projects_data {
-        let mut authorized = !project_data.status.is_hidden();
-
-        if let Some(user) = &user_option {
-            if !authorized {
-                if user.role.is_mod() {
-                    authorized = true;
-                } else {
-                    let user_id: database::models::ids::UserId = user.id.into();
-
-                    let project_exists = sqlx::query!(
-                            "SELECT EXISTS(SELECT 1 FROM team_members WHERE team_id = $1 AND user_id = $2)",
-                            project_data.inner.team_id as database::models::ids::TeamId,
-                            user_id as database::models::ids::UserId,
-                        )
-                        .fetch_one(&**pool)
-                        .await?
-                        .exists;
-
-                    authorized = project_exists.unwrap_or(false);
-                }
-            }
-        }
-
-        if authorized {
-            projects.push(convert_project(project_data));
+    // FIXME: async functions don't work in iterators
+    let mut projects = Vec::with_capacity(projects_data.len());
+    for data in projects_data {
+        if is_authorized(&data, &user_option, &pool).await? {
+            projects.push(Project::from(data))
```

Comment:
You don't have to, but using future joins making these parallelized would be awesome